src/lib.rs

```diff
@@ -455,7 +455,30 @@ impl AffinePoint {
     /// Attempts to interpret a byte representation of an
     /// affine point, failing if the element is not on
     /// the curve or non-canonical.
-    pub fn from_bytes(mut b: [u8; 32]) -> CtOption<Self> {
+    pub fn from_bytes(b: [u8; 32]) -> CtOption<Self> {
+        Self::from_bytes_inner(b, 1.into())
+    }
+
+    /// Attempts to interpret a byte representation of an affine point, failing if the
+    /// element is not on the curve.
+    ///
+    /// Most non-canonical encodings will also cause a failure. However, this API
+    /// preserves (for use in consensus-critical protocols) a bug in the parsing code that
+    /// caused two non-canonical encodings to be silently accepted:
+    ///
+    /// - (0, 1), which is the identity;
+    /// - (0, -1), which is a point of order two.
+    ///
+    /// Each of these has a single non-canonical encoding in which the value of the sign
+    /// bit is 1.
+    ///
+    /// See [ZIP 216](https://zips.z.cash/zip-0216) for a more detailed description of the
+    /// bug, as well as its fix.
+    pub fn from_bytes_with_zip_216_bug(b: [u8; 32]) -> CtOption<Self> {
```

Comment:
I agree with @daira that we should rename this (I don't mind zir suggestion too.) It's not a bug necessarily, but it's also not a bug in ZIP 216.

Reply:
Fair, `from_bytes_with_bug_fixed_by_zip216` would be more accurate than the current name.